subt/tools/submit.py

```diff
@@ -0,0 +1,155 @@
+import argparse
+import json
+import re
+import requests
+import subprocess
+import os
+import pathlib
+
+import toml
+
+WORLDS=dict(
+    up1 = "Urban Practice 1",
+    up2 = "Urban Practice 2",
+    up3 = "Urban Practice 3",
+    cq  = "Cave Qualification",
+    cs1 = "Cave Simple 1",
+    cs2 = "Cave Simple 2",
+    cs3 = "Cave Simple 3",
+    cp1 = "Cave Practice 1",
+    cp2 = "Cave Practice 2",
+    cp3 = "Cave Practice 3",
+)
+
+ROBOTS=dict(
+    teambase = "TEAMBASE",
+    drone = "SSCI_X4_SENSOR_CONFIG_2",
+    freya = "ROBOTIKA_FREYJA_SENSOR_CONFIG_2",
+)
+
+
+def _cmd(*args):
+    return subprocess.run(args, stdout=subprocess.PIPE).stdout
+
+
+def validate_world(world):
+    try:
+        return WORLDS[world]
+    except KeyError:
+        sys.exit(f"'{world}' not valid world identification")
+
+
+def validate_robots(robots, timeout):
+    valid = {}
+    for name, kind in robots.items():
+        try:
+            valid[name] = ROBOTS[kind]
+        except KeyError:
+            sys.exit(f"'{kind}' not valid robot kind identification")
+    valid[f"T{timeout}"] = "TEAMBASE"
+    return valid
+
+
+def validate_image(image):
+    image_bytes = bytes(image, 'ascii')
+    images = _cmd('aws', 'ecr', 'list-images', '--repository-name', 'subt/robotika')
+    for img in images.split(b'\n'):
+        parts = img.split(b'\t')
+        if parts[-1] == image_bytes:
+            return f'200670743174.dkr.ecr.us-east-1.amazonaws.com/subt/robotika:{image}'
+    sys.exit(f"image '{image}' not found in aws subt/robotika repository")
+
+
+def generate_simname(logdir, image, world):
+    if not logdir.exists():
```

Comment:
this is probably specific to your setup? I.e. if I analyze logfiles on different computer then where this tool is used then it will be always `rc1`

Reply:
it worked (I started 2nd simulation with Freyja) so now I wonder how is it possible ;). One thing is that `logdir` is rather dir where is `run.toml` file ... i.e. in my case it is not related to logs. I see now, it is creating new directories so you should have all config files in one place corresponding to future logfiles. OK (a bit unexpected but OK)

Reply:
p.s. this is all written in your description (which I did not read properly), so maybe copy and paste some parts to the script itself? (about directory?)

Reply:
It needs to somehow keep track of already generated names. I haven't found a way how to list all existing simulation runs.